### PR TITLE
Clean up setup.py, _version.py, and remove stale environment.yml

### LINKED ISSUE #40: Symmetrize edge vs node normalization
## Summary

Edge features are systematically disadvantaged relative to node features due to normalization asymmetries at two levels:

1. **Model input level** Edge embeddings bypass `input_norm` and `input_dropout`
2. **GTConv layer level** Edge path uses post-norm while node path uses pre-norm

These compound across stacked layers, progressively dampening edge feature variance.

---

## Edge embeddings bypass input normalization and dropout

**File:** `gt_pyg/nn/model.py` (`GraphTransformerNet.forward()`)

### Current (asymmetric)
```
node_dim_in -> [node_emb] -> [input_norm] -> [input_dropout] -> GT layers
edge_dim_in -> [edge_emb] -------------------------------------> GT layers
```

### Proposed (symmetric)
```
node_dim_in -> [node_emb] -> [input_norm]    -> [input_dropout] -> GT layers
edge_dim_in -> [edge_emb] -> [edge_inp_norm] -> [input_dropout] -> GT layers
```

### Changes
- Add `self.edge_input_norm` (same type as `input_norm` -- BatchNorm1d or LayerNorm) in `__init__`
- Apply `self.edge_input_norm(e)` and `self.input_dropout(e)` after `self.edge_emb(edge_attr)` in `forward()`

---

## Edge post-norm vs node pre-norm asymmetry in GTConv

**File:** `gt_pyg/nn/gt_conv.py` (`GTConv.forward()`)

### Current (asymmetric)
```
NODES:  x_res -> norm1 -> Q,K,V -> attn -> WO -> drop -> (+x_res) -> norm2 -> FFN -> drop -> (+) -> x_out
                pre-norm                                               pre-norm

EDGES:  e_res -> (no norm) -> _eij -> WOe -> drop -> (+e_res) -> norm1e -> FFN_e -> drop -> (+) -> norm2e -> edge_out
                                                                 pre-norm                          POST-norm (extra!)
```

Edges have:
- No pre-norm before the attention output projection
- An extra post-norm (`norm2e`) on exit that nodes don't have

### Proposed (symmetric pre-norm)
```
NODES:  x_res -> norm1 -> Q,K,V -> attn -> WO -> drop -> (+x_res) -> norm2 -> FFN -> drop -> (+) -> x_out
                pre-norm                                               pre-norm

EDGES:  e_res -> norm1e -> _eij -> WOe -> drop -> (+e_res) -> norm2e -> FFN_e -> drop -> (+) -> edge_out
                pre-norm                                      pre-norm            (no post-norm)
```

### Changes
- Move `norm1e` to before the edge attention projection (pre-norm for attention, mirroring `norm1` for nodes)
- Rename current `norm1e` to `norm2e` (pre-FFN norm, mirroring `norm2` for nodes)
- Remove the final `norm2e` post-norm on edge output

---

## Files to modify
- `gt_pyg/nn/model.py` -- add `edge_input_norm`, apply norm + dropout to edge embeddings
- `gt_pyg/nn/gt_conv.py` -- restructure edge norm to match node pre-norm pattern

## Notes
- This is an architectural change that will affect trained model weights -- existing checkpoints will not be compatible
- Should be validated with training experiments to confirm improvement

### PR DESCRIPTION
**Pull request by pgniewko**
## Summary

- **setup.py**: Fix file handle leak — wrap `open()` in a `with` statement
- **setup.py**: Remove `pandas` from `install_requires` — no source file imports it
- **_version.py**: Fix tag parsing fragility — use regex to handle hyphenated tags (e.g. `v1.0.0-rc1`) instead of `rsplit("-", 2)` which breaks on them
- **data/utils.py**: Remove 9 unused imports from `atom_features` (already exported by `data/__init__.py`)
- **environment.yml**: Delete stale conda export dump (py37 build strings, duplicate rdkit entries, pinned `gt-pyg==0.0.0`)

Closes #40 
## Test plan

- [x] All 75 tests pass (`pytest gt_pyg/ -v`)
- [x] `import gt_pyg` succeeds (version derivation still works)
- [x] `from gt_pyg.data.utils import get_tensor_data` succeeds (unused imports removed, used ones kept)

### CODE REVIEW

File: gt_pyg/nn/gt_conv.py [hunk 1]
```diff
@@ -310,11 +315,10 @@ def forward(self, x, edge_index, edge_attr=None):
             e_attn = self.dropout_layer(e_attn)
 
             e1 = edge_res + e_attn  # residual
-            e1_norm = self.norm1e(e1)
+            e1_norm = self.norm1e(e1)  # pre-norm before FFN
             e_ffn = self.ffn_e(e1_norm)
             e_ffn = self.dropout_layer(e_ffn)
-            e2 = e1 + e_ffn
-            edge_out = self.norm2e(e2)
+            edge_out = e1 + e_ffn  # residual, no trailing norm (matches node path)
```
Comment (pgniewko) [inline gt_pyg/nn/gt_conv.py]: This has been addressed in the previous PR

File: gt_pyg/nn/gt_conv.py [hunk 2]
```diff
@@ -59,6 +59,8 @@ def __init__(
 
         super().__init__(node_dim=0, aggr=aggr)
 
+        if num_heads <= 0:
```
Comment (pgniewko) [inline gt_pyg/nn/gt_conv.py]: isn't that already addressed?

File: gt_pyg/nn/tests/test_gt_conv.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+"""Tests for GTConv normalization symmetry (#40) and input validation (#30)."""
+
+import torch
+import pytest
+
+from gt_pyg.nn.gt_conv import GTConv
+
+
+class TestNormalizationSymmetry:
```
Comment (pgniewko) [inline gt_pyg/nn/tests/test_gt_conv.py]: Make sure these are up to date, and align with the gt_conv implementation after mergin PR#43
